packages/app/src/app/overmind/effects/vscode/SandboxFsSync/index.ts

```diff
@@ -404,30 +404,33 @@ class SandboxFsSync {
     autoInstallTypes: boolean
   ) {
     try {
-      if (
-        autoInstallTypes &&
-        this.typesInfo[dep.name] &&
-        !dep.name.startsWith('@types/')
-      ) {
-        const name = `@types/${dep.name}`;
-        this.fetchDependencyTypingFiles(name, this.typesInfo[dep.name].latest)
-          .then(files => {
-            this.setAndSendPackageTypes(dep.name, files);
-          })
-          .catch(e => {
-            if (process.env.NODE_ENV === 'development') {
-              console.warn('Trouble fetching types for ' + name);
-            }
-          });
-      }
-
       try {
         const files = await this.fetchDependencyTypingFiles(
           dep.name,
           dep.version
         );
-
-        this.setAndSendPackageTypes(dep.name, files);
+        const hasTypes =
+          Object.keys(files).filter(key => key.endsWith('.d.ts')).length > 0;
```

Comment:
One check, maybe we need to check for the dependency path when checking (so something like `key.startsWith(dep.name) && key.endsWith('.d.ts')` because we tend to also include `.d.ts` files of transient dependencies sometimes, even if the dependency itself doesn't have the types.

We could also use `.find` or `.some` here, then it would stop traversing the files as soon as it found one match!

Reply:
OH, yes! Very nice feedback, updated 😄